modules/holodeckb2b-core/src/test/java/org/holodeckb2b/ebms3/handlers/inflow/AuthorizeMessageTest.java

```diff
@@ -0,0 +1,191 @@
+
+/**
+ * Created at 23:44 29.01.17
+ *
+ * @author Timur Shakuov (t.shakuov at gmail.com)
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class AuthorizeMessageTest {
+
+    @Mock
+    private Appender mockAppender;
+    @Captor
+    private ArgumentCaptor captorLoggingEvent;
+
+    private static String baseDir;
+
+    private static HolodeckB2BTestCore core;
+
+    private AuthorizeMessage handler;
+
+    @BeforeClass
+    public static void setUpClass() throws Exception {
+        baseDir = AuthorizeMessageTest.class.getClassLoader()
+                .getResource("handlers").getPath();
+        core = new HolodeckB2BTestCore(baseDir);
+        HolodeckB2BCoreInterface.setImplementation(core);
+    }
+
+    @Before
+    public void setUp() throws Exception {
+        handler = new AuthorizeMessage();
+        LogManager.getRootLogger().addAppender(mockAppender);
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        LogManager.getRootLogger().removeAppender(mockAppender);
+    }
+
+    @Test
+    public void testDoProcessing() throws Exception {
+        MessageMetaData mmd = TestUtils.getMMD("handlers/full_mmd.xml", this);
+        // Creating SOAP envelope
+        SOAPEnvelope env = SOAPEnv.createEnvelope(SOAPEnv.SOAPVersion.SOAP_12);
+        // Adding header
+        SOAPHeaderBlock headerBlock = Messaging.createElement(env);
+        // Adding UserMessage from mmd
+        OMElement umElement = UserMessageElement.createElement(headerBlock, mmd);
+//        System.out.println("umElement: " + umElement);
+        MessageContext mc = new MessageContext();
+        mc.setFLOW(MessageContext.IN_FLOW);
+        // Setting input message property
+        mc.setProperty(MessageContextProperties.IN_USER_MESSAGE, umElement);
+        try {
+            mc.setEnvelope(env);
+        } catch (AxisFault axisFault) {
+            fail(axisFault.getMessage());
+        }
+
+        PMode pmode = new PMode();
+        pmode.setMep(EbMSConstants.ONE_WAY_MEP);
+        pmode.setMepBinding(EbMSConstants.ONE_WAY_PUSH);
+
+        UserMessage userMessage
+                = UserMessageElement.readElement(umElement);
+
+        String pmodeId = userMessage.getCollaborationInfo().getAgreement().getPModeId();
+
+        // todo It seems strange that we need to set the PMode id value separately
+        // todo when it is contained within the agreement
+        // todo But if we don't set it the value returned by userMessage.getPModeId() is null now
+        userMessage.setPModeId(pmodeId);
+
+        String msgId = userMessage.getMessageId();
+
+        pmode.setId(pmodeId);
+
+        core.getPModeSet().add(pmode);
+
+        // Setting token configuration
+        UsernameTokenConfig tokenConfig = new UsernameTokenConfig();
+        tokenConfig.setUsername("username");
+        tokenConfig.setPassword("secret");
+
+        mc.setProperty(SecurityConstants.EBMS_USERNAMETOKEN, tokenConfig);
+
+        // Setting signature configuration
```

Comment:
As the signature configuration is not needed for this test I would remove it to improve readability of the test.